apps/vaporgui/FlowSubtabs.cpp

```diff
@@ -497,29 +497,40 @@ void FlowSeedingSubtab::_resizeFlowParamsVectors() {
     // Going from 3d vectors to 2d vectors.
     // Save the values we remove for restoration later on.
     if ( _numDims == 2 ) {  
-        _oldZRakeNumSeeds = rakeSeeds[Z];
-        rakeSeeds.resize(2);
-
-        _oldZRakeMin = rakeRegion[Z_RAKE_MIN];
-        _oldZRakeMax = rakeRegion[Z_RAKE_MAX];
-        rakeRegion.resize(4);
-
-        _oldZPeriodicity = periodicity[Z];
-        periodicity.resize(2);
+        if ( rakeSeeds.size() != 2 ) {
```

Comment:
This logic, as well as in the 'if' clauses below, assumes that rakeSeeds is either of size 2 or 3. Otherwise it will generate an invalid memory access. A more robust solution would avoid that assumption (consider the possibility of a 1D rake). At the very least asserts should be included to ensure this assumption is true.

Reply:
Please let me know if you think the current changes are sufficient.